comp/core/workloadmeta/collectors/util/kube_metadata_util.go

```diff
@@ -0,0 +1,40 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+// Package util contains utility functions for image metadata collection
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// GenerateKubeMetadataEntityID generates and returns a unique entity id for KubernetesMetadata entity
+// for namespaced objects, the id will have the format {resourceType}/{namespace}/{name} (e.g. deployments/default/app )
+// for cluster scoped objects, the id will have the format {resourceType}//{name} (e.g. node//master-node)
+func GenerateKubeMetadataEntityID(resource, namespace, name string) string {
```

Comment:
Should we define a new type like `type KubeMetadataEntityID string` and return it here?

We could also use it in the params of workloadmeta `GetKubernetesMetadata` so it's easier to understand what it expects and that callers are expected to call this function to get the ID.